### PR TITLE
feat: estimate fn call action without loading cost

### PR DESCRIPTION
**Pull request by jakmeier**
`action_function_call_*` costs are currently used to also cover contract
loading. But we want to give `wasm_contract_loading_*` costs meaningful
values and consequently reduce `action_function_call_*`.

This adjusts the `ActionFunctionCallBase` estimation to not include
contract loading costs. The difference is rather singificant since the
test contract used before was almost 100kB in size.

This change also introduces an approach of subtracting receipt overhead
completely, by executing multiple actions into one receipt.

### CODE REVIEW

**Comment by jakmeier** [comment]
FYI: Time-based estimation for `ActionFunctionCallBase` is reduced from ~380Ggas to ~100Ggas.

**Comment by matklad** [comment]
>The difference is rather singificant since the
test contract is almost 100kB.

can we use a trivial contract instead?

**Comment by matklad** [comment]
Was too terse :) 

Let's see if we can get the same numeric result by just using a trivial contract. I'd feel more comfortable with that, as relying on the empty method names feels hacky. 

Alternatively, we can just subtract base loading cost perhaps? Or would that subtract too much?

If neither of the ideas work, I am fine with the current impl as well! (if we add one more agument to `_ex` though, lets switch to builder.) Given that the difference is significant, we should probably do this.

**Comment by jakmeier** [comment]
This pulled me down another estimator rabbit-hole once again. See #7003 and ##7002. I think these should be resolved first before deciding here how to move forward.

**Comment by jakmeier** [comment]
@matklad Took me more than a month but finally I have resolved all surrounding issues. I think I improved both the estimation code and the accuracy of the results. 

Time results give me ~35 Gas on gcloud, while the `icount` based estimation for `ActionFunctionCallBase` is around 20 Ggas. In other words, the overhead of launching a contract, once all the receipt and other DB overhead is removed, is only around 20-35us! 

### Details on new estimation approach
So now in my latest attempt, the estimation works without the empty-method-hack. Instead using a trivial contract and subtracting the executable loading base.

Additionally, I discovered a new approach to getting rid of the nasty overhead. The other issues I resolved have shown me, just how big even the per-receipt overhead is. So I decided that for this case, where we want the pure action cost, it is better to put many actions into the same receipt. This means only one account pair needs to be loaded from the database, only one outcome is produced, and so on. And then I subtract the cost of a transaction that has the same kind of action in it, but only one of them. The difference is the pure action cost times (n-1).

In theory, this could be done for all action estimations. It might actually help with some of the high variance observed there. But the problem is that if the action has a side-effect, then this is not a good way to measure it. Because we need to also cover the cost of the side-effect. And doing the same action on the same receiver over and over again, in most case, leads to side-effects shadowing each other. For example, transferring N times between the same two accounts leads to the same amount of DB writes, as we only update the balance in the DB at the end of the chunk processing.

So actually, this trick currently only applies to Fn Call costs.

**Comment by matklad** [comment]
>it is better to put many actions into the same receipt.

This is very, very good!

File: runtime/runtime-params-estimator/src/utils.rs [hunk 1]
```diff
@@ -233,26 +237,47 @@ pub(crate) fn fn_cost_in_contract(
...
+    let block_size = 1;
     let (gas_cost, _ext_costs) =
-        aggregate_per_block_measurements(ctx.config, block_size, measurements, Some(overhead));
-    gas_cost
+        aggregate_per_block_measurements(ctx.config, block_size, measurements, overhead);
+    gas_cost.saturating_sub(&base_gas_cost, &NonNegativeTolerance::Strict) / (n_actions - 1) as u64
+}
+
+fn function_call_action(method_name: String) -> Action {
+    Action::FunctionCall(near_primitives::transaction::FunctionCallAction {
```
Comment (matklad) [inline runtime/runtime-params-estimator/src/utils.rs:275]:
```suggestion
    Action::FunctionCall(FunctionCallAction {
```

I think actions are intended to be imported

File: runtime/runtime-params-estimator/src/lib.rs [hunk 2]
```diff
@@ -660,11 +657,15 @@ fn pure_deploy_bytes(ctx: &mut EstimatorContext) -> GasCost {
     (cost_4mb - cost_empty) / (large_code_len - small_code_len) as u64
 }
 
+/// Base cost for a fn call action, without receipt creation or contract loading.
 fn action_function_call_base(ctx: &mut EstimatorContext) -> GasCost {
-    let total_cost = noop_function_call_cost(ctx);
-    let base_cost = action_sir_receipt_creation(ctx);
-
-    total_cost.saturating_sub(&base_cost, &NonNegativeTolerance::PER_MILLE)
+    let inner_iterations = 100;
```
Comment (matklad) [inline runtime/runtime-params-estimator/src/lib.rs:662]: Perhaps just `n_actions` to not have two different names here?

File: runtime/runtime-params-estimator/src/utils.rs [hunk 3]
```diff
@@ -208,22 +208,26 @@ pub(crate) fn fn_cost_with_setup(
...
-    let n_blocks = ctx.config.warmup_iters_per_block + ctx.config.iter_per_block;
+    let n_warmup_blocks = ctx.config.warmup_iters_per_block;
+    let n_blocks = n_warmup_blocks + ctx.config.iter_per_block;
     let mut testbed = ctx.testbed();
 
-    let chosen_accounts = {
+    let mut chosen_accounts = {
         let tb = testbed.transaction_builder();
-        std::iter::repeat_with(|| tb.random_unused_account()).take(n_blocks).collect::<Vec<_>>()
+        std::iter::repeat_with(|| tb.random_unused_account()).take(n_blocks + 1).collect::<Vec<_>>()
```
Comment (matklad) [inline runtime/runtime-params-estimator/src/utils.rs:228]: I usually `use std::iter`

**Review by matklad — APPROVED** [review]

File: runtime/runtime-params-estimator/src/utils.rs (hunk 1 above)
Comment (jakmeier) [reply runtime/runtime-params-estimator/src/utils.rs]: done

File: runtime/runtime-params-estimator/src/lib.rs (hunk 2 above)
Comment (jakmeier) [reply runtime/runtime-params-estimator/src/lib.rs]: done

**Pull request merged**